### PR TITLE
show DSR-dispatch mode in cilium-dbg status

### LINKED ISSUE #29132: kpr: show DSR-dispatch mode in cilium-dbg status
When running `cilium-dbg status --verbose` on a cluster with [KPR in DSR mode](https://docs.cilium.io/en/latest/network/kubernetes/kubeproxy-free/#direct-server-return-dsr), it would be great to also show the selected DSR dispatch mode (IP-option, Geneve, IPIP) in the output. See below for an example of the current output.

```
KubeProxyReplacement Details:
  Status:                 Strict
  Socket LB:              Enabled
  Socket LB Coverage:     Full
  Socket LB Protocols:    TCP, UDP
  Devices:                eth0 10.242.0.4 (Direct Routing)
  Mode:                   SNAT
  Backend Selection:      Random
  Session Affinity:       Enabled
  Graceful Termination:   Enabled
  NAT46/64 Support:       Disabled
  XDP Acceleration:       Disabled
  Services:
  - ClusterIP:      Enabled
  - NodePort:       Enabled (Range: 30000-32767)
  - LoadBalancer:   Enabled
  - externalIPs:    Enabled
  - HostPort:       Enabled 
```

### PR DESCRIPTION
**Pull request by chaunceyjiang**
Fixes: #29132

```release-note
show DSR-dispatch mode in cilium-dbg status
```

### CODE REVIEW

**Comment by chaunceyjiang** [comment]
image

File: pkg/client/client.go [hunk 1]
```diff
@@ -632,6 +632,9 @@ func FormatStatusResponse(w io.Writer, sr *models.StatusResponse, sd StatusDetai
 			}
 			xdp = np.Acceleration
 			mode = np.Mode
+			if mode == models.KubeProxyReplacementFeaturesNodePortModeDSR {
+				dsrMode = np.DsrMode
+			}
```
Comment (julianwiedmann) [inline pkg/client/client.go]:
```suggestion
			if mode == models.KubeProxyReplacementFeaturesNodePortModeDSR ||
			mode == models.KubeProxyReplacementFeaturesNodePortModeHybrid
			{
				dsrMode = np.DsrMode
			}
```

Hybrid does a mix of DSR and SNAT, so the DSR-dispatch mode also becomes relevant here.

File: api/v1/openapi.yaml [hunk 2]
```diff
@@ -2376,6 +2376,12 @@ definitions:
                 - SNAT
                 - DSR
                 - Hybrid
+              dsrMode:
+                type: string
+                enum:
+                  - opt
+                  - ipip
+                  - geneve
```
Comment (julianwiedmann) [inline api/v1/openapi.yaml]:
```suggestion
                  - IP Option/Extension
                  - IPIP
                  - Geneve
```

Let's pretty-print it :)

File: pkg/client/client.go [hunk 3]
```diff
@@ -698,6 +701,9 @@ func FormatStatusResponse(w io.Writer, sr *models.StatusResponse, sd StatusDetai
 		if mode != "" {
 			fmt.Fprintf(tab, "  Mode:\t%s\n", mode)
 		}
+		if dsrMode != "" {
+			fmt.Fprintf(tab, "    DSR Mode:\t%s\n", dsrMode)
```
Comment (julianwiedmann) [inline pkg/client/client.go]:
```suggestion
			fmt.Fprintf(tab, "    DSR Dispatch Mode:\t%s\n", dsrMode)
```

Does this still format properly, or is the line becoming too wide?

**Review by julianwiedmann — CHANGES_REQUESTED** [review]
Thank you for tackling this! Just a few small change suggestions below.

File: pkg/client/client.go (hunk 1 above)
Comment (chaunceyjiang) [reply pkg/client/client.go]: Updated.

File: api/v1/openapi.yaml (hunk 2 above)
Comment (chaunceyjiang) [reply api/v1/openapi.yaml]: Updated.

**Comment by chaunceyjiang** [comment]
image

**Review by julianwiedmann — APPROVED** [review]
I'm happy, thank you @chaunceyjiang !

(@aanm , it's all yours)

**Review by aanm — APPROVED** [review]

**Pull request merged**